integ/src/nodegroup_provider.rs

```diff
@@ -0,0 +1,573 @@
+        .resource_type(ResourceType::Instance)
+        .tags(
+            Tag::builder()
+                .key("Name")
+                .value(format!("{}_node", cluster_name))
+                .build(),
+        )
+        .tags(
+            Tag::builder()
+                .key(format!("kubernetes.io/cluster/{}", cluster_name))
+                .value("owned")
+                .build(),
+        )
+        .tags(
+            Tag::builder()
+                .key(INSTANCE_TAG_NAME)
+                .value(INSTANCE_TAG_VALUE)
+                .build(),
+        )
+        .build()
+}
+
+fn userdata(endpoint: &str, cluster_name: &str, certificate: &str) -> String {
+    base64::encode(format!(
+        r#"[settings.updates]
+            ignore-waves = true
+                
+            [settings.kubernetes]
+            api-server = "{}"
+            cluster-name = "{}"
+            cluster-certificate = "{}"
+        "#,
+        endpoint, cluster_name, certificate
+    ))
+}
+
+async fn wait_for_conforming_nodegroup(
+    eks_client: &aws_sdk_eks::Client,
+    cluster_name: &str,
+    action: &str,
+    nodegroup_name: &str,
+) -> ProviderResult<()> {
+    loop {
+        if !non_conforming_nodegroup(eks_client, cluster_name, action, nodegroup_name).await? {
+            tokio::time::sleep(Duration::from_millis(1000)).await;
+            continue;
+        }
+        return Ok(());
+    }
+}
+
+async fn non_conforming_nodegroup(
+    eks_client: &aws_sdk_eks::Client,
+    cluster_name: &str,
+    action: &str,
+    nodegroup_name: &str,
+) -> ProviderResult<bool> {
+    match action {
+        "create" => {
+            // let nodegroup_status = get_nodegroup_status(eks_client, cluster_name).await?;
+            let nodegroup_status = eks_client
+                .describe_nodegroup()
+                .nodegroup_name(nodegroup_name)
+                .cluster_name(cluster_name)
+                .send()
+                .await
+                .context("Unable to describe nodegroup")?
+                .nodegroup
+                .context("Unable to extract nodegroup")?
+                .status
+                .context("Unable to extract nodegroup status")?;
+            match nodegroup_status {
+                NodegroupStatus::Active => Ok(true),
+                _ => Ok(false),
+            }
+        }
+        "delete" => confirm_nodegroup_exists(eks_client, cluster_name, nodegroup_name).await,
+        _ => return Err(ProviderError::new_with_context("Invalid action input")),
+    }
+}
+
+async fn confirm_nodegroup_exists(
+    eks_client: &aws_sdk_eks::Client,
+    cluster_name: &str,
+    nodegroup_name: &str,
+) -> ProviderResult<bool> {
+    let nodegroup = eks_client
+        .describe_nodegroup()
+        .nodegroup_name(nodegroup_name)
+        .cluster_name(cluster_name)
+        .send()
+        .await;
+
+    match nodegroup {
+        Err(_resource_not_found_exception) => Ok(true),
+        _ => Ok(false),
```

Comment:
Are these bools reversed? I would expect `resource_not_found` to result in `false`.

Reply:
I think the function name `confirm_nodegroup_exists` is not accurate, and I'll rename it to `confirm_nodegroup_deleted`. thanks